python/source/flight.rst

```diff
@@ -605,3 +605,102 @@ Or if we use the wrong credentials on login, we also get an error:
     server.shutdown()
 
 .. _(HTTP) basic authentication: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme
+
+Authentication with certificates
+=================================
+
+Following on from the previous scenario where traffic to the server is managed via a username and password, 
+HTTPS (more specifically TLS) communication allows an additional layer of security by encrypting messages
+between the client and server. This is achieved using certificates. During development, the easiest 
+approach is developing with self-signed certificates. At startup, the server loads the public and private 
+key and the client client authenticates itself to the server with a public key.
+
+.. note:: In production environments it is recommended to make use of a certificate signed by a certificate authority.
+.. note:: This tutorial uses Windows to create a self-signed certificate. For Linux environments, other methods such as OpenSSL can be used.
```

Comment:
If possible, we should drop this note, and just link to other resources.

Alternatively, we can remove the instructions altogether, and point to the testing data repository. "Generate a self-signed certificate by using `dotnet` on Windows (link), or `openssl` on Linux or MacOS (link). Or, use the self-signed certificate from the Arrow testing data repository (link)." https://github.com/apache/arrow-testing/tree/master/data/flight